### PR TITLE
fix an edge case of BandPass probability propagation

### LINKED ISSUE #5765: Wrong center locations in BandPassActivityProfile
### Affected tool(s) or class(es)
org.broadinstitute.hellbender.utils.activityprofile.BandPassActivityProfile

### Affected version(s)
Latest master branch as of March 6, 2019

### Description 
Line 147 should read:
```Java
final SimpleInterval loc = getLocForOffset(superState.getLoc(), i)
```
instead of
```Java
final SimpleInterval loc = getLocForOffset(justAddedState.getLoc(), i)
```

Likewise, line 154 should read:
```Java
states.add(superState)
```
instead of
```Java
states.add(justAddedState)
```

Otherwise, the location changes in org.broadinstitute.hellbender.utils.activityprofile.ActivityProfile, line 241 doesn't make a lot of sense, and neither does the extension of `getMaxProbPropagationDistance()` in org.broadinstitute.hellbender.utils.activityprofile.BandPassActivityProfile, lines 100-102.

### PR DESCRIPTION
**Pull request by davidbenjamin**
Closes #5765.

@jamesemery would you mind looking at this?

### CODE REVIEW

**Comment by jamesemery** [comment]
First, it looks like you need to address the failing haplotype caller tests now: https://storage.googleapis.com/hellbender-test-logs/build_reports/master_25237.4/tests/test/index.html

File: src/main/java/org/broadinstitute/hellbender/utils/activityprofile/BandPassActivityProfile.java [hunk 1]
```diff
@@ -144,14 +144,14 @@ public double getSigma() {
         for ( final ActivityProfileState superState : super.processState(justAddedState) ) {
             if ( superState.isActiveProb() > 0.0 ) {
                 for( int i = -filterSize; i <= filterSize; i++ ) {
-                    final SimpleInterval loc = getLocForOffset(justAddedState.getLoc(), i);
+                    final SimpleInterval loc = getLocForOffset(superState.getLoc(), i);
                     if ( loc != null ) {
                         final double newProb = superState.isActiveProb() * gaussianKernel[i + filterSize];
```
Comment (jamesemery) [inline src/main/java/org/broadinstitute/hellbender/utils/activityprofile/BandPassActivityProfile.java]: Why in the world is this method still called isActiveProb() when it returns a double? I suppose this isn't your problem but this code could use some work

File: src/main/java/org/broadinstitute/hellbender/utils/activityprofile/BandPassActivityProfile.java [hunk 2]
```diff
@@ -144,14 +144,14 @@ public double getSigma() {
         for ( final ActivityProfileState superState : super.processState(justAddedState) ) {
             if ( superState.isActiveProb() > 0.0 ) {
                 for( int i = -filterSize; i <= filterSize; i++ ) {
-                    final SimpleInterval loc = getLocForOffset(justAddedState.getLoc(), i);
```
Comment (jamesemery) [inline src/main/java/org/broadinstitute/hellbender/utils/activityprofile/BandPassActivityProfile.java]: We should probably expand the test under `ActivityProfileUnitTest.testSoftClips()` to be extensive enough to catch this edge case. Maybe we should hard code some expected probability distributions or add a line asserting that the underlying simple locations for each state is not overlapping.

**Comment by droazen** [comment]
@davidbenjamin @jamesemery As discussed in person, we should consider the possibility that this behavior is by design, and the band pass filter is deliberately being centered on the base before the high-quality soft clip begins. Let's talk about this before hitting merge on this one.

**Comment by davidbenjamin** [comment]
@droazen @jamesemery It does seem like it's by design, and as James found the behavior is to multiply the evidence by the length of the soft clip, which seems weird.  We avoid this in Mutect2 and treat it like an indel.  I would vote for doing the same and just eliminating the `HIGH_QUALITY_SOFT_CLIP` category.

**Comment by droazen** [comment]
@davidbenjamin How would you like to proceed on this one?

**Comment by davidbenjamin** [comment]
@droazen I'm for closing this PR and instead opening one that gets rid of the dubious `HIGH_QUALITY_SOFT_CLIP` behavior.

**Comment by jamesemery** [comment]
@davidbenjamin I'm going to go ahead and close this. I summarized the discussion and questions this raised in a new issue lest we forget about this problem in the future: #5897.

**Pull request closed without merging**